modules/perf-metrics/observer.js

```diff
@@ -0,0 +1,70 @@
+import { getLogger } from 'jitsi-meet-logger';
+
+import { createPerfMetricsEvent } from '../../service/statistics/AnalyticsEvents';
+import Statistics from '../statistics/statistics';
+
+import { JITSI_METRIC_PREFIX } from './constants';
+
+const logger = getLogger(__filename);
+
+/**
+ * Reference to the singleton performance observers.
+ */
+let observer;
+
+/**
+ * Initializes a {@link PerformanceObserver} for logging performance metrics.
+ */
+export function initialize() {
+    if ('PerformanceObserver' in window) {
+        if (observer) {
+            dispose();
+        }
+        observer = new PerformanceObserver((list, observer_) => {
+            const metrics = [];
+
+            list.getEntries().forEach(entry => {
+                if (observer === observer_ && entry.name.startsWith(JITSI_METRIC_PREFIX)) {
+                    metrics.push(entry);
+                    logger.debug(`${entry.name.replace(JITSI_METRIC_PREFIX, '')} took ${entry.duration}ms`);
```

Comment:
It will be useful to include the time from navigation start to the end marker as well.

For example during our recent performance testing we were interested in how much time does it take from the navigation start to moment when the session initiate packet is received.

Reply:
Do you think that would be useful wholesale or for specific metrics?